analysis/grammar_search_analysis.py

```diff
@@ -0,0 +1,146 @@
+"""Debugging script for grammar search invention approach.
```

Comment:
maybe let's move analysis/ into src/ so that mypy looks at it? we can exclude it from coverage requirements

Reply:
(by adding an exclusion rule for `src/analysis/` somehow)